pkg/controller/complianceremediation/complianceremediation_controller.go

```diff
@@ -378,6 +395,123 @@ func (r *ReconcileComplianceRemediation) countKubeUnmetDependencies(rem *compv1a
+	_, isRequiredValuesProcessed := labels[compv1alpha1.RemediationValueRequiredProcessedLabel]
+	var unsetRequireValueList []string
+	var setRequireValueList []string
+
+	//Go through values required list to find is it inside the tailored profile
+	for _, requiredValue := range requiredValuesList {
+		if requiredValue == "" {
+			continue
+		}
+		found, err := r.isRequiredValueSet(rem, requiredValue)
+		if err != nil {
+			return hasUpdate, reconcile.Result{}, fmt.Errorf("Error finding if required value is set: %v", err)
+		}
+		if found {
+			setRequireValueList = append(setRequireValueList, requiredValue)
+		} else {
+			unsetRequireValueList = append(unsetRequireValueList, requiredValue)
+		}
+
+	}
+
+	currentUnsetValues := strings.Split(annotations[compv1alpha1.RemediationUnsetValueAnnotation], ",")
+	currentUsedValues := strings.Split(annotations[compv1alpha1.RemediationValueUsedAnnotation], ",")
+	if !isRequiredValuesProcessed {
+		logger.Info("Updating remediation to denote values have been processed")
+		labels[compv1alpha1.RemediationValueRequiredProcessedLabel] = ""
+		if len(unsetRequireValueList) > 0 {
+			annotations[compv1alpha1.RemediationUnsetValueAnnotation] = strings.Join(append(currentUnsetValues, unsetRequireValueList...), ",")
+		}
+		if len(setRequireValueList) > 0 {
+			annotations[compv1alpha1.RemediationValueUsedAnnotation] = strings.Join(append(currentUsedValues, setRequireValueList...), ",")
+		}
+		logger.Info("Labeling remediation to denote some values haven't been set")
+		rCopy := rem.DeepCopy()
+		rCopy.SetAnnotations(annotations)
+		rCopy.SetLabels(labels)
+		err := r.client.Update(context.TODO(), rCopy)
+		if err != nil {
+			return hasUpdate, reconcile.Result{}, fmt.Errorf("adding xccdf required-value label/annotation: %w", err)
+		}
+		hasUpdate = true
+		return hasUpdate, reconcile.Result{}, nil
+	}
+	return hasUpdate, reconcile.Result{}, nil
+}
+
+//To find if it is set in tailored profile
+func (r *ReconcileComplianceRemediation) isRequiredValueSet(rem *compv1alpha1.ComplianceRemediation, requiredValue string) (bool, error) {
+	var scan = &compv1alpha1.ComplianceScan{}
+	err := r.client.Get(context.TODO(), types.NamespacedName{
+		Namespace: rem.GetNamespace(),
+		Name:      rem.GetScan(),
+	}, scan)
+
+	if err != nil {
+		return false, err //error getting scan
+	}
+	if scan.Spec.TailoringConfigMap != nil {
+		tp := &compv1alpha1.TailoredProfile{}
+		err = r.client.Get(context.TODO(), types.NamespacedName{Name: scan.Spec.TailoringConfigMap.Name, Namespace: rem.GetNamespace()}, tp)
+		if err != nil {
+			return false, err
+		}
+		for _, setValue := range tp.Spec.SetValues {
+			if setValue.Name == requiredValue {
+				return true, nil
+			}
+		}
+		return false, nil
+	}
+	//scan did not have a tailored profile
+	return false, nil
+}
+
+func (r *ReconcileComplianceRemediation) handleUnsetValues(rem *compv1alpha1.ComplianceRemediation, logger logr.Logger) (bool, reconcile.Result, error) {
```

Comment:
Do we need to return the `reconcile.Result` variable from this function? it seems like it's set to `reconcile.Result{}` in all branches?
Note that the reason that `handleUnmetDependencies` returns the `reconcile.Result` type is that in some branches, that function returns `reconcile.Result{Requeue: true, RequeueAfter: defaultDependencyRequeueTime}`, IOW, reconcile, but with a delay.

Reply:
That's a good catch, we can remove that return as well!